src/common/inc/CaPTkGUIUtils.h

```diff
@@ -311,10 +311,25 @@ inline std::string getCaPTkDataDir()
   return captk_dataDir;
 }
 
-//! opens the link using Qt's desktop services
+//! opens the link using Qt's desktop services 
 inline bool openLink(const std::string &link)
 {
-  return QDesktopServices::openUrl(QUrl(link.c_str()));
+#ifndef _WIN32
+  // temporarily remove CaPTk libs from LD_LIBRARY_PATH to avoid interference with desktop apps
+  // vars set by AppRun in linux/mac appimage
+  std::string captk_ldpathValue = cbica::getEnvironmentVariableValue("LD_LIBRARY_PATH");
+  std::string desktop_ldpathValue = cbica::getEnvironmentVariableValue("ORIGINAL_LD_LIBRARY_PATH");
+  if (desktop_ldpathValue.length() > 0)
```

Comment:
Can we put this in an if-block, perhaps? That way we can preserve the functionality for older distros:

```cpp
bool result = QDesktopServices::openUrl(QUrl(link.c_str()));
if (!result)
{
  // do the fancy stuff with environment variable here
  // call the openUrl function again
}
```

Reply:
We will need to test the return value in different use cases as it may return true but still fail to open the url. As per the documentation: 

> Warning: A return value of true indicates that the application has successfully requested the operating system to open the URL in an external application. The external application may still fail to launch or fail to open the requested URL. This result will not be reported back to the application.

Reply:
I think in the case Mark was facing, the result value would actually have returned "true" -- it successfully requested the OS to open that URL, and it tried to do this using kde-open.

Reply:
But really, if an external application registered to open URL schema fails without access to CaPTk's libs, the user has severe configuration issues with their machine. It isn't our responsibility to ensure their external link-opening application works, we just need to make sure it's not our changes to the LD_LIBRARY_PATH that break it. This is true for older distros as well -- none of their desktop apps should be relying on our libs. That they haven't complained so far is just a lucky break.

Reply:
Oh, nice catch, @ashishsingh18. Anyway, I was proposing doing this only in specific cases and not for all Linux distros throughout, as will run into the opposite issue with older distros, where there might be a *very* old Qt in the system paths.